Src/AwesomeAssertions/Primitives/StringEndStrategy.cs

```diff
@@ -19,24 +20,60 @@ public StringEndStrategy(IEqualityComparer<string> comparer, string predicateDes
 
     public void ValidateAgainstMismatch(AssertionChain assertionChain, string subject, string expected)
     {
-        assertionChain
-            .ForCondition(subject!.Length >= expected.Length)
-            .FailWith($"{ExpectationDescription}{{0}}{{reason}}, but {{1}} is too short.", expected, subject);
-
-        if (!assertionChain.Succeeded)
+        if (expected.Length is 0 || comparer.Equals(subject, expected))
         {
             return;
         }
 
-        int indexOfMismatch = subject.Substring(subject.Length - expected.Length).IndexOfFirstMismatch(expected, comparer);
+        var indexes = IndexOfLastMismatch(subject, expected, comparer);
+
+        bool hasMismatch = expected.Length > subject.Length || indexes.index2 >= 0;
```

Comment:
`index.index2` is difficult to understand: "What means 'index1' vs. 'index2'?"
I see several options to solve that:
1. `IndexOfLastMismatch` seems to be used only once. So, you can make it more specific and name input and output according the real usage.
2. Deconstruct `var indexes` into `(int mismatchInSubject, int mismatchInExpecation)`.